packages/typescript-estree/src/convert.ts

```diff
@@ -993,6 +997,10 @@ export class Converter {
         if (parent.kind === SyntaxKind.ObjectLiteralExpression) {
           method.params = node.parameters.map(el => this.convertChild(el));
 
+          if (method.type === AST_NODE_TYPES.TSEmptyBodyFunctionExpression) {
```

Comment:
This is required b/c `Property#value` cannot be of type `TSEmptyBodyFunctionExpression`, so we can't pass `method` [here](https://github.com/typescript-eslint/typescript-eslint/pull/1289/files#diff-68b2b0b0d2d0ff1846b1114472008332R1007) if it's `TSEmptyBodyFunctionExpression`.

This is a bit of a strawman - The tests pass, and going over it in my head if I'm understanding everything correctly it should never throw, so I don't think `Property` needs its type changed.

However, I'm not sure what the best way to represent the relationship for TS to be happy, aside from either asserting like this, or adding it to the if condition - both are "ok" in my books, but open to suggestions if there's a better or more prefered way :)

Reply:
I'm not sure exactly why this code is parsed without errors
```
a={ get() }
```
but looks like it might cause these exception.

Reply:
A winner is you!

~I've not run it in the tests, but looking at the AST in AST ~

You bet me to it 😄

Reply:
That's weird, I would have thought that typescript would hard error for us in this case: [ts playground](https://www.typescriptlang.org/play/#code/MYewdgzgLgBAhjAvDA3jAZiEAKAlDAXyA)

Reply:
Yeah, TypeScript just [parses it as a method body](https://astexplorer.net/#/gist/d8840821b2458e2a105061634e9c0334/78b5846bc3a6048d13b9b570a9db16c27fd459a8) apparently.

Reply:
With `--allowJs`, TS [successfully parses](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=1&pc=27&useJavaScript=true#code/MYewdgzgLgBAhjAvDA3jAZiEAKAlAbgBoYAjOAJxgF98g) even this so to say JavaScript code:
```ts
const a = { foo();, bar };
```
It's full of surprises.

Reply:
For the second crash, I've raised #1301 to track it.
It's some weird syntax that we'd never encountered/considered before.